File: src/control_stick_viz.ts
import { SlippiGame } from './slippi'
import { Coord, CheckResult, Violation } from './index';

export function controlStickViz(game: SlippiGame, playerIndex: number, coords: Coord[]): CheckResult {
    let coordViolation: Violation = new Violation(0, "Control Stick Viz", coords)
    return new CheckResult(false, [coordViolation])
}

File: libenforcer-wasm/src/checks/control_stick_viz.rs
use crate::types::{CheckResult, Coord, Violation};

/// Control stick visualization "check" — packages all coordinates as evidence
/// for visualization purposes. Always returns result: false (no violation).
pub fn check(coords: &[Coord]) -> CheckResult {
    let violation = Violation::with_evidence(
        0.0,
        "Control Stick Viz".to_string(),
        coords.to_vec(),
    );
    CheckResult {
        result: false,
        violations: vec![violation],
    }
}
